mava/adders/reverb/base.py

```diff
@@ -114,6 +140,151 @@ def __init__(
         )
         self._use_next_extras = use_next_extras
 
+    def write_experience_to_tables(  # noqa
+        self,
+        trajectory: Union[Trajectory, mava_types.Transition],
+        table_priorities: Dict[str, Any],
+    ) -> None:
+        # Get a dictionary of the transition nets and agents.
+        if self._table_network_config:
+            entry_extras = trajectory.extras["network_int_keys"]
+            entry_net_keys = {}
+            agents = sort_str_num(trajectory.actions.keys())
+            for agent in agents:
+                arr = entry_extras[agent].numpy()
+                if type(trajectory) == Step:
+                    entry_net_keys[agent] = self._int_to_nets[arr[0]]
+                else:
+                    entry_net_keys[agent] = self._int_to_nets[arr]
+
+            # Get the unique agents and mapping from net_keys to agents.
+            agents, trans_nets_agent = get_trans_net_agents(
+                trajectory=trajectory, entry_net_keys=entry_net_keys
+            )
+
+        # Check if experience was used
+        created_item = False
+
+        # If all the network entries of a trainer is in the data then add it to that
+        # trainer's data table.
+        for table, priority in table_priorities.items():
+            if self._table_network_config is None:
+                created_item = True
+                self._writer.create_item(
+                    table=table, priority=priority, trajectory=trajectory
+                )
+            else:
+                # Check if all the networks are in trans_nets_agent.
+                trans_dict_copy = copy.deepcopy(trans_nets_agent)
+
+                # While the networks are in the data keep creating tables
+                # Each training example can therefore create multiple items
+                is_in_entry = True
+                while is_in_entry:
+                    item_agents = []
+                    for net_key in self._table_network_config[table]:
+                        if (
+                            net_key in trans_dict_copy
+                            and len(trans_dict_copy[net_key]) > 0
+                        ):
+                            item_agents.append(trans_dict_copy[net_key].pop())
+                        else:
+                            is_in_entry = False
+                            break
+
+                    if is_in_entry:
+                        created_item = True
+
+                        # Create new empty transition
+                        if type(trajectory) == Step:
+                            soe = trajectory.start_of_episode  # type: ignore
```

Comment:
Do we need all the `type: ignore` here? There are quite a few in the file. Perhaps better to ignore the entire file for now? But I guess would be better to fix the type issues.

Reply:
Yes, I would ideally like to fix the type: ignores. The problem is that trajectory can either be a Step or Transition which means it has different properties. mypy does not seem to pick up that I do a check for Step and therefore still complains that if it is a Transition line 200 will fail. I am not sure how one can fix this? @KaleabTessera any ideas?